tests/testAutopas/tests/containers/octree/OctreeTests.cpp

```diff
@@ -0,0 +1,791 @@
+
+  // Insert particles, the first should not cause the octree to split
+  srand(1234);
+  for (int unsigned i = 0; i < treeSplitThreshold; ++i) {
+    auto particle = getRandomlyDistributedParticle(min, max);
+    root->insert(root, particle);
+    ASSERT_FALSE(root->hasChildren());
+  }
+
+  // These should cause the octree to split
+  for (int unsigned i = 0; i < treeSplitThreshold; ++i) {
+    auto particle = getRandomlyDistributedParticle(min, max);
+    root->insert(root, particle);
+    ASSERT_TRUE(root->hasChildren());
+  }
+}
+
+#if !defined(AUTOPAS_OPENMP)
+std::tuple<OctreeTest::Vector3DList, OctreeTest::Vector3DList,
+           std::vector<std::tuple<unsigned long, unsigned long, double>>>
+OctreeTest::calculateForcesAndPairs(autopas::ContainerOption containerOption, autopas::TraversalOption traversalOption,
+                                    autopas::DataLayoutOption dataLayoutOption, autopas::Newton3Option newton3Option,
+                                    size_t numParticles, size_t numHaloParticles, std::array<double, 3> boxMin,
+                                    std::array<double, 3> boxMax, double cellSizeFactor, bool doSlightShift,
+                                    double cutoff, double skin, double interactionLength,
+                                    Vector3DList particlePositions, Vector3DList haloParticlePositions) {
+  using namespace autopas;
+
+  double _cutoffsquare = cutoff * cutoff;
+  static constexpr double _eps{1.};
+  static constexpr double _sig{1.};
+
+  // Construct container
+  ContainerSelector<Molecule> selector{boxMin, boxMax, cutoff};
+  selector.selectContainer(
+      containerOption, autopas::ContainerSelectorInfo{cellSizeFactor, skin, 32, autopas::LoadEstimatorOption::none});
+  auto container = selector.getCurrentContainer();
+
+  // Create a functor that is able to calculate forces
+  autopas::LJFunctor<Molecule, true /*applyShift*/, false /*useMixing*/, autopas::FunctorN3Modes::Both,
+                     false /*calculateGlobals*/>
+      ljFunctor{cutoff};
+  ljFunctor.setParticleProperties(_eps * 24, _sig * _sig);
+
+  for (int unsigned i = 0; i < numParticles; ++i) {
+    auto position = particlePositions[i];
+    auto particle = Molecule(position, {0, 0, 0}, i);
+    container->addParticle(particle);
+  }
+
+  for (int unsigned i = 0; i < numHaloParticles; ++i) {
+    auto position = haloParticlePositions[i];
+    auto particle = Molecule(position, {0, 0, 0}, numParticles + i);
+    container->addHaloParticle(particle);
+  }
+
+  // Obtain a compatible traversal
+  auto traversal =
+      autopas::utils::withStaticCellType<Molecule>(container->getParticleCellTypeEnum(), [&](auto particleCellDummy) {
+        return autopas::TraversalSelector<decltype(particleCellDummy)>::generateTraversal(
+            traversalOption, mockFunctor, container->getTraversalSelectorInfo(), dataLayoutOption, newton3Option);
+      });
+
+  if (doSlightShift) {
+    // TODO(johannes): Shift the particles
```

Comment:
This is also an artifact from adopting the code from the TraversalComparison test. However, since this is not part of the scope that is tested here, I removed this in f2b59fc. The slight-shifting capabilities of the container are tested in the TraversalComparison test and do not need to be tested here.